modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -437,6 +437,11 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyht
 	reqHost := clonedReq.Host
 	reqHeader := clonedReq.Header
 
+	var bufferedReqBody []byte
```

Comment:
We can modify the `Close` behavior of `bodyReadCloser`. Instead of putting the buffer to the pool, just close the body if not nil. Using `defer` below to put the buffer to the pool, so there is no need to copy the bytes.